flocker/acceptance/test_environment_variables.py

```diff
@@ -166,19 +170,13 @@ def mysql_can_connect():
                     try:
                         return connect(
                             host=host,
-                            port=MYSQL_EXTERNAL_PORT,
+                            port=port,
                             user=user,
                             passwd=passwd,
                             db=db,
                         )
-                    except OperationalError as e:
-                        # PyMySQL doesn't provided a structured way to
-                        # get this.
-                        # https://github.com/PyMySQL/PyMySQL/issues/274
-                        if "Connection refused" in str(e):
-                            return False
-                        else:
-                            raise
+                    except Error:
```

Comment:
This is in fact kinda fragile and brittle, and ideally we'll replace them with tests that don't involve arbitrary complexity like mysql clients. But given we're not replacing them immediately, I think this is probably best we can do.